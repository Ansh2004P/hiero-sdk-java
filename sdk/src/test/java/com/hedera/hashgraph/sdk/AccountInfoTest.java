package com.hedera.hashgraph.sdk;

import com.hedera.hashgraph.sdk.proto.CryptoGetInfoResponse;
import com.hedera.hashgraph.sdk.proto.Response;
import io.github.jsonSnapshot.SnapshotMatcher;
import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;


public class AccountInfoTest {
    private static final PrivateKey privateKey = PrivateKey.fromString(
        "302e020100300506032b657004220420db484b828e64b2d8f12ce3c0a0e93a0b8cce7af1bb8f39c97732394482538e10");

    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    @Test
    void fromProtobuf() {
        Response response = Response.newBuilder()
            .setCryptoGetInfo(
                CryptoGetInfoResponse.newBuilder()
                    .setAccountInfo(CryptoGetInfoResponse.AccountInfo.newBuilder()
                        .setKey(privateKey.getPublicKey().toKeyProtobuf()))
            )
            .build();

        SnapshotMatcher.expect(AccountInfo.fromProtobuf(response.getCryptoGetInfo().getAccountInfo()).toString())
            .toMatchSnapshot();
    }
}
