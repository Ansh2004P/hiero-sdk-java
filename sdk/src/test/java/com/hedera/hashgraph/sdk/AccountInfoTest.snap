com.hedera.hashgraph.sdk.AccountInfoTest.fromProtobuf=[
  "AccountInfo{accountId=0.0.0, contractAccountId=, deleted=false, proxyAccountId=null, proxyReceived=0 tℏ, key=302a300506032b6570032100e0c8ec2758a5879ffac226a13c0c516b799e72e35141a0dd828f94d37988a4b7, balance=0 tℏ, sendRecordThreshold=0 tℏ, receiveRecordThreshold=0 tℏ, receiverSignatureRequired=false, expirationTime=1970-01-01T00:00:00Z, autoRenewPeriod=PT0S}"
]